packages/ckeditor5-dev-env/lib/release-tools/tasks/update-package-versions.js

```diff
@@ -0,0 +1,232 @@
+ *
+ * @param {String} pathToUpdate Directory containing files to update.
+ * @param {Boolean} dryRun If set to true, diff of changes that would be made is calculated, and included in the returned object. Without
+ * this flag, file is updated normally.
+ * @returns {Object} Object containing numerical values counting files found and updated, as well as array of all changes made. If `dryRun`
+ * is set to true, `differences` array will be empty, as there is no point in calculating that which is only used during `dryRun`.
+ */
+function updateDirectory( pathToUpdate, dryRun ) {
+	const globPattern = pathToUpdate + '/*/package.json';
+	const packageJsonArray = glob.sync( globPattern );
+
+	if ( !packageJsonArray.length ) {
+		return { found: 0, updated: 0 };
+	}
+
+	let updatedFiles = 0;
+	const differences = [];
+
+	for ( const file of packageJsonArray ) {
+		const currentFileData = fs.readFileSync( file, 'utf-8' );
+		const parsedData = JSON.parse( currentFileData );
+
+		updateObjectProperty( parsedData, 'dependencies' );
+		updateObjectProperty( parsedData, 'devDependencies' );
+
+		const newFileData = JSON.stringify( parsedData, null, 2 ) + '\n';
+
+		if ( currentFileData !== newFileData ) {
+			updatedFiles++;
+
+			if ( dryRun ) {
+				differences.push( {
+					file,
+					content: diff( currentFileData, newFileData, { newlineIsToken: true } )
+				} );
+			} else {
+				fs.writeFileSync( file, newFileData, 'utf-8' );
+			}
+		}
+	}
+
+	return { found: packageJsonArray.length, updated: updatedFiles, differences };
+}
+
+/**
+ * This function takes an object and a property name, and modifies said property in that object. If any of the properties
+ * of that property matches the regex, its value will be updated to the value of `version` property from the root of the object.
+ *
+ * @param {Object} parsedPkgJson Object to update.
+ * @param {String} propertyName Name of the property to update.
+ */
+function updateObjectProperty( parsedPkgJson, propertyName ) {
+	// Update only the CKEditor 5 dependencies, except the *-dev and *-inspector.
+	const regex = /^@ckeditor\/ckeditor5-(?!dev|inspector)|^ckeditor5$/;
+	const version = parsedPkgJson.version;
+
+	for ( const dependency in parsedPkgJson[ propertyName ] ) {
+		if ( !regex.test( dependency ) ) {
+			continue;
+		}
+
+		parsedPkgJson[ propertyName ][ dependency ] = `^${ version }`;
+	}
+}
+
+/**
+ * Takes in raw changelog for a single file generated by jsdiff library, and returns formatted array of strings, containing
+ * human-readable, line by line changelog of a file, with removals and additions colored, as well as unnecessary chunks of
+ * unchanged file hidden.
+ *
+ * @param {Array<Object>} diff Array of changes for a single file generated by jsdiff library.
+ * @returns {Array<String>} Formatted changelog split into single lines.
+ */
+function formatDiff( diff ) {
+	const formattedDiff = [];
+	const regex = /(?<=":) (?=")/;
```

Comment:
Why this regexp is needed?

Reply:
I need to split these lines and merge them back together in order to pretty print the diff. Indentations are also kept in these lines, so there are more than one space. I suppose i could split using last space of a string instead, but this seems safer.

```json
    "@ckeditor/ckeditor5-core": "^31.1.0",
```

Reply:
What if, for whatever reason, the script changes something else in the `package.json` file, other than the CKEditor 5 dependency versions? Now, only changes to the dependency versions are displayed in red and green colors, but other changes that could occur in the file are not highlighted in the console.

The `--dry-run` flag should display all the changes that the script has made, to be sure that running it without this flag will not surprise us.

If the dependency version has changed, it can be highlighted as it is now (only the version, not the full line). But when something else has changed that is not expected, let the entire changed line be highlighted.

Reply:
Okay, i've actually taken this into account, but then i forgot to color the rest of changes, so good thing You pointed that out 😅

Single removals and additions are simple enough, they are just colored lines. But how do we want to show replacements?

a) Replacement on the same line

![](https://user-images.githubusercontent.com/69513154/147705820-b93e2da0-9e92-4b98-a299-32dabf61f9f0.png)

![](https://user-images.githubusercontent.com/69513154/147706400-e9e288b6-eea5-4f35-862f-891a83a79e9d.png)

b) Replacement on the same line, but trim the whitespaces off the addition

![](https://user-images.githubusercontent.com/69513154/147705888-00e0ea00-36f2-41ff-bdf5-b02e47628936.png)

![](https://user-images.githubusercontent.com/69513154/147706377-472c2e70-d591-482f-9470-f34484056767.png)

c) Replacement on the line below 

![](https://user-images.githubusercontent.com/69513154/147705952-833e51b0-a25b-4a1b-9a3c-62231673fbcd.png)

![](https://user-images.githubusercontent.com/69513154/147706449-0c16c275-d8d8-4634-8be3-8c0f09d19629.png)

I'd suppose option b) is preferable, so this is what I'll do for now.